src/cargo/core/manifest.rs

```diff
@@ -384,7 +384,10 @@ impl Target {
         self
     }
     pub fn set_harness(&mut self, harness: bool) -> &mut Target {
-        self.harness = harness;
+        match  self.kind {
+            TargetKind::Lib(_) => (),
+            _ => self.harness = harness,
+        }
```

Comment:
If `harness == false` then there is no binary produced. I have not been able to find out why that is, but the error goes similar to the linked issue.